src/ducks/export/services.js

```diff
@@ -0,0 +1,163 @@
+  return format({
+    delimiter: ';',
+    quoteColumns: true,
+    headers: [
+      'Date',
+      'Realisation date',
+      'Assigned date',
+      'Label',
+      'Original bank label',
+      'Category name',
+      'Amount',
+      'Currency',
+      'Type',
+      'Reimbursement status',
+      'Bank name',
+      'Account name',
+      'Account number',
+      'Account type',
+      'Recurrence name',
+      'Tag 1',
+      'Tag 2',
+      'Tag 3',
+      'Tag 4',
+      'Tag 5'
+    ]
+  })
+}
+
+/**
+ * Generator that transforms the given transactions into our own CSV format and yields the result line by line.
+ *
+ * @param {Array<Transaction>} transactions The list of transactions to transform
+ *
+ * @return {void}
+ */
+export const transactionsToCSV = function* (transactions) {
+  for (const tr of transactions) {
+    const account = tr.account.data
+    const recurrence = tr.recurrence.data
+    const tags = tr.tags.data
+
+    const data = [
+      dateStr(tr.date),
+      dateStr(tr.realisationDate),
+      dateStr(tr.applicationDate),
+      tr.label,
+      tr.originalBankLabel,
+      categories[tr.cozyCategoryId],
+      tr.amount,
+      tr.currency,
+      tr.type,
+      tr.reimbursementStatus
+    ]
+
+    if (account != null) {
+      data.push(
+        account.institutionLabel,
+        account.label,
+        account.number,
+        account.type
+      )
+    } else {
+      data.push(undefined, undefined, undefined)
+    }
+
+    if (recurrence != null) {
+      data.push(recurrence.manualLabel || recurrence.automaticLabel)
+    } else {
+      data.push(undefined)
+    }
+
+    if (tags != null) {
+      for (let i = 0; i < 5; i++) {
+        if (tags[i] != null) {
+          data.push(tags[i].label)
+        } else {
+          data.push(undefined)
+        }
+      }
+    } else {
+      data.push(undefined, undefined, undefined, undefined, undefined)
+    }
+
+    yield data
+  }
+}
+
+/**
+ * Upload the given content to cozy-stack as a file with the given name in the given directory.
+ *
+ * @param {CozyClient} client A CozyClient instance
+ * @param {Blob|Stream|string|ArrayBuffer} data Content to be uploaded
+ * @param {object} options
+ * @param {string} options.name Name of the uploaded file
+ * @param {string} options.dirId Id of the directory where the file will be uploaded to
+ *
+ *
+ * @returns {Promise<object>} Created io.cozy.files
+ * @throws {Error} explaining the reason why the creation failed
+ *
+ */
+export const uploadFile = async (
+  client,
+  data,
+  { name = 'export-cozy-banks.csv', dirId = ROOT_DIR_ID }
+) => {
+  const files = client.collection(FILES_DOCTYPE)
+
+  try {
+    return await files.createFile(data, {
+      name,
+      dirId
+    })
+  } catch (err) {
+    if (err.status !== 409) {
+      throw err
+    }
+
+    // Trash the conflicting file
+    const { data: existing } = await files.statByPath(`/${name}`)
+    await files.destroy(existing)
```

Comment:
Is it really what we want to put it in the trash? Don't we want to remove it at all? 

Is there a reason to not "update" the file in order to create a version? (see for instance https://github.com/cozy/cozy-client/blob/master/packages/cozy-client/src/models/file.js#L461-L473)

Reply:
For some reason I didn't think about updating the file. This seems like the best strategy.
I wasn't aware of the existence of `uploadFileWithConflictStrategy` but it completely serves my purpose so I'll use it instead of creating my own version :).